web/views.py

```diff
@@ -22,6 +23,20 @@ def get_context_data(self, **kwargs):
 
 class AdminPanelView(TemplateView):
     template_name = 'web/admin_panel.html'
+    possible_permissions = [
+        "news.add_article", "news.change_article", "news.delete_article",
+        "news.add_event", "news.change_event", "news.delete_event",
+        "news.add_timeplace", "news.change_timeplace", "news.delete_timeplace",
+        "make_queue.can_create_event_reservation",
+        "make_queue.change_quota",
+        "make_queue.change_printer3dcourse",
+        "groups.can_edit_group",
+    ]
+
+    def get_context_data(self, **kwargs):
```

Comment:
Shouldn't this be done with a `PermissionRequiredMixin` or `UserPassesTestMixin`?

Reply:
Also, this is technically a 403 error

Reply:
`PermissionRequiredMixin` does not work in this situation, as somebody should have access to the admin panel even though they do not have access to all the admin subsystems. Never seen `UserPassesTestMixin` before, but seems to be a good fit. Also totally agree that it should be 403.